### PR TITLE
Remove uses of NLsolve

### PR DESCRIPTION
**Pull request by avik-pal**

### CODE REVIEW

File: src/initialize_dae.jl [hunk 1]
```diff
@@ -20,13 +20,12 @@ function BrownFullBasicInit(; abstol = 1e-10, nlsolve = nothing)
...
 
-using SciMLNLSolve
 default_nlsolve(alg, isinplace, u, autodiff = false) = alg
 function default_nlsolve(::Nothing, isinplace, u, autodiff = false)
-    NLSolveJL(autodiff = autodiff ? :forward : :central)
+    TrustRegion(; autodiff = autodiff ? AutoForwardDiff() : AutoFiniteDiff())
 end
 function default_nlsolve(::Nothing, isinplace::Val{false}, u::StaticArray, autodiff = false)
-    SimpleNewtonRaphson(autodiff = autodiff)
+    SimpleNewtonRaphson(autodiff = autodiff ? AutoForwardDiff() : AutoFiniteDiff())
```
Comment (avik-pal) [inline src/initialize_dae.jl:28]: @ChrisRackauckas should we change this to `SimpleTrustRegion` to make it consistent with the non static array version?

Comment (ChrisRackauckas) [reply src/initialize_dae.jl:28]: Yes

**Comment by avik-pal** [comment]
failures seem to be same as master

File: src/initialize_dae.jl [hunk 2]
```diff
@@ -564,7 +563,7 @@ function _initialize_dae!(integrator, prob::DAEProblem,
     if alg.nlsolve !== nothing
         nlsolve = alg.nlsolve
     else
-        nlsolve = NewtonRaphson(autodiff = isAD)
+        nlsolve = NewtonRaphson(autodiff = alg_autodiff(integrator.alg))
```
Comment (ChrisRackauckas) [inline src/initialize_dae.jl:566]: We should keep it as type information now that we can use it effectively.

Comment (ChrisRackauckas) [reply src/initialize_dae.jl:566]: Yes it changes the type level stuff to a bool as backwards compat support IIRC

Comment (avik-pal) [reply src/initialize_dae.jl:566]:
```
function alg_autodiff(alg)
    autodiff = _alg_autodiff(alg)
    if autodiff == Val(false)
        return AutoFiniteDiff()
    elseif autodiff == Val(true)
        return AutoForwardDiff()
    else
        return _unwrap_val(autodiff)
    end
end
```
This should constant propagate and give the correct AD type right?

Comment (ChrisRackauckas) [reply src/initialize_dae.jl:566]: We shouldn't rely on constant propagation. It can fail. But let's pull this in and make this better in a follow up.

**Pull request merged**